File: src/main/java/gg/generations/rarecandy/tools/gui/Selected.java
package gg.generations.rarecandy.tools.gui;

public class Selected {
    private final RareCandyCanvas canvas;
    private String variant;
    private int variantId = -1;
    private String mesh;
    private int meshId = -1;

    public Selected(RareCandyCanvas canvas) {
        this.canvas = canvas;
    }


    public void setVariant(String variant) {
        this.variant = variant;
        this.variantId = -1;

        if (variant == null || canvas.loadedModel == null) return;

        this.variantId = canvas.loadedModel.variantNameToId.getOrDefault(variant, -1);
        if (variantId != -1 && canvas.loadedModelInstance != null) {
            canvas.loadedModelInstance.setVariant(variantId);
        }
    }

    public void setMesh(String mesh) {
        this.mesh = mesh;
        this.meshId = -1;

        if (mesh == null || canvas.loadedModel == null) return;

        this.meshId = canvas.loadedModel.meshNameToId.getOrDefault(mesh, -1);
    }

    public String getVariant() {
        return variant;
    }

    public int getVariantId() {
        return variantId;
    }

    public String getMesh() {
        return mesh;
    }

    public int getMeshId() {
        return meshId;
    }
}

File: src/main/java/gg/generations/rarecandy/tools/gui/Selected.kt
package gg.generations.rarecandy.tools.gui

import gg.generations.rarecandy.renderer.components.MultiRenderObject
import gg.generations.rarecandy.renderer.rendering.ObjectInstance

object Selected {
    var variant: String? = null
        set(variant) {
            field = variant
            this.variantId = -1

            if (variant == null || model == null) return

            this.variantId = model!!.variantNameToId.getOrDefault(variant, -1)
            if (variantId != -1 && instance != null) {
                instance!!.setVariant(variantId)
            }
        }
    var variantId: Int = -1
        private set
    var mesh: String? = null
        set(mesh) {
            field = mesh
            this.meshId = -1

            if (mesh == null || model == null) return

            this.meshId = model!!.meshNameToId.getOrDefault(mesh, -1)
        }
    var meshId: Int = -1
        private set

    private val model: MultiRenderObject?
        get() = ModelTracker.model

    private val instance: ObjectInstance?
        get() = ModelTracker.instance
}
